cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitor.java

```diff
@@ -500,6 +547,43 @@ public ClusterModel clusterModel(long from,
     return clusterModel;
   }
 
+  /**
+   * Get replica placement information, i.e. each replica resides on which disk of the broker.
+   *
+   * @param clusterModel The cluster model to populate replica placement information.
+   * @param cluster The cluster metadata.
+   * @return A map from topic partition to replica placement information.
+   *
+   */
+  private Map<TopicPartition, Map<Integer, String>> getReplicaPlacementInfo(ClusterModel clusterModel, Cluster cluster) {
+    Map<TopicPartition, Map<Integer, String>> replicaPlacementInfo = new HashMap<>();
+    Map<Integer, KafkaFuture<Map<String, LogDirInfo>>> logDirsByBrokerId =
+        _adminClient.describeLogDirs(cluster.nodes().stream().mapToInt(Node::id).boxed().collect(Collectors.toList())).values();
+    for (Map.Entry<Integer, KafkaFuture<Map<String, LogDirInfo>>> entry : logDirsByBrokerId.entrySet()) {
+      Integer brokerId =  entry.getKey();
+      try {
+        entry.getValue().get(LOGDIR_RESPONSE_TIMEOUT_MS, TimeUnit.MILLISECONDS).forEach((logdir, info) -> {
+          if (info.error == Errors.NONE) {
+            for (Map.Entry<TopicPartition, ReplicaInfo> e : info.replicaInfos.entrySet()) {
+              if (!e.getValue().isFuture) {
+                replicaPlacementInfo.putIfAbsent(e.getKey(), new HashMap<>());
+                replicaPlacementInfo.get(e.getKey()).put(brokerId, logdir);
+              }
```

Comment:
Nit: Can we add `LOG.trace(...)` in a new `else` block to trace in progress logdir creation?